collated/main/A0144727B.md

```diff
@@ -0,0 +1,1123 @@
+    public static final String MESSAGE_INVALID_TASK_FIELD = "The task field provided is invalid";
+    public static final String MESSAGE_TASKS_LISTED_OVERVIEW = "%1$d tasks listed!";
+    public static final String MESSAGE_NO_START_DATE_SPECIFIED = "No start date specified";
+    public static final String MESSAGE_NO_END_DATE_SPECIFIED = "No end date specified";
+    public static final String MESSAGE_NO_START_TIME_SET = "No start time set";
+    public static final String MESSAGE_NO_END_TIME_SET = "No end time set";
+    public static final String MESSAGE_NO_PREVIOUS_COMMAND = "No previous command";
+    public static final String MESSAGE_NO_PREVIOUS_UNDO_COMMAND = "No previous undo command";
+    public static final String MESSAGE_DEST_FILE_EXISTS = "Specified file already exists";
+    public static final String MESSAGE_INVALID_TASK_VALUE = "The value for task name provided is invalid";
+    public static final String MESSAGE_INVALID_DATE_TIME_VALUE = "The value for date or time provided is invalid";
+    public static final String MESSAGE_INVALID_TIME_VALUE = "The value for time provided is invalid";
+    public static final String MESSAGE_INVALID_PRIORITY_VALUE = "The priority value is invalid";
+    public static final String MESSAGE_NO_PRIORITY_VALUE = "No priority level set";
+}
+```
+###### /main/java/seedu/ggist/commons/events/ui/ChangeListingEvent.java
+``` java
+public class ChangeListingEvent extends BaseEvent {
+
+    public final String listing;
+    
+    public ChangeListingEvent(String listing) {
+        this.listing = listing;
+    }
+
+    @Override
+    public String toString() {
+        if (listing == null || listing.equals("")) {
+            return (new Date()).toString();
+        } else {
+            return listing;
+        }
+    }
+}
+```
+###### /main/java/seedu/ggist/logic/commands/DoneCommand.java
+``` java
+public class DoneCommand extends Command {
+
+    public static final String COMMAND_WORD = "done";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Marks the task identified by the index number used in the current listing as done.\n"
+            + "Parameters: INDEX (must be a positive integer)\n" + "Example: " + COMMAND_WORD + " 1";
+
+    public static final String MESSAGE_DONE_TASK_SUCCESS = "Task Done: %1$s";
+
+    public final ArrayList<Integer> targetIndexes;
+
+    public DoneCommand(ArrayList<Integer> indexes) {
+        this.targetIndexes = indexes;
+    }
+    
+```
+###### /main/java/seedu/ggist/model/ModelManager.java
+``` java
+    @Override
+    public synchronized void doneTask(ReadOnlyTask target) throws TaskNotFoundException {
+        taskManager.doneTask(target);
+        updateListing();
+        indicateTaskManagerChanged();
+    }
+
+    public synchronized void editTask(ReadOnlyTask target, String field, String value) throws TaskNotFoundException, IllegalValueException {
```

Comment:
this line is too long requiring horizontal scrolling; please split it appropriately